src/adagrams.js

```diff
@@ -1,15 +1,61 @@
+import { LETTER_SCORE, LETTER_POOL } from './gameConstants';
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letters = [];
+  const letterPool = Object.keys(LETTER_POOL);
+  while (letters.length < 10) {
+    const random = Math.floor(Math.random() * letterPool.length);
+    const addLetter = letterPool[random];
+    if (letters.filter(x => x === addLetter).length < LETTER_POOL[addLetter]) {
+      letters.push(addLetter);
+    }
+  }
+  return letters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const word = input.toUpperCase();
+  const availableLetters = {};
+  lettersInHand.forEach(item => {
+    if (availableLetters[item]) { availableLetters[item]++; }
+    else { availableLetters[item] = 1; }
+  });
```

Comment:
Nice frequency map for the letters. Consider making a helper that builds a frequency map, which could apply both to the hand and the word.